test_scripts/Resumption/Handling_errors_from_HMI/commonResumptionErrorHandling.lua

```diff
@@ -1647,4 +1784,57 @@ function m.reRegisterAppsCustom_AnotherRPC(pTimeToRegApp2, pRPC)
   end
 end
 
+--[[ @isSubscribed: send OnInteriorVehicleData
+--! @parameters:
+--! pAppId - application number (1, 2, etc.)
+--! pIsExpApp1 - true(default) - if it's expected notification on mobile app1
+--! pIsExpApp2 - true - if it's expected notification on mobile app2
+--! pModuleType - module type to subscribe for
+--! pModuleId - module id to subscribe for
+--! @return: none
+--]]
+function m.isSubscribed(pIsExpApp1, pIsExpApp2, pModuleType, pModuleId)
+  if pIsExpApp1 == nil then pIsExpApp1 = true end
+  pModuleType = pModuleType or m.defaultModuleType
+  pModuleId = pModuleId or m.getModuleControlData(pModuleType,1).moduleId
+  rc.rc.isSubscribed(pModuleType, pModuleId, 1, pIsExpApp1)
+  if pIsExpApp2 ~= nil then
+    local occurences2 = pIsExpApp2 == true and 1 or 0
+    local params = m.getActualModuleIVData(pModuleType, pModuleId)
+    m.getMobileSession(2):ExpectNotification("OnInteriorVehicleData", { moduleData = params }):Times(occurences2)
+  end
+end
+
+--[[ @geInteriorVDvalue: get value for interior vehicle data with moduleType, moduleId, subscribe parameters
+--! @parameters:
+--! pModuleType - module type value to define in structure
+--! pSubscribe - subscribe value to define in structure
+--! pModuleId - module id value to define in structure
+--! @return: built structure
+--]]
+function m.geInteriorVDvalue(pModuleType, pSubscribe, pModuleId)
+  pModuleId = pModuleId or m.getModuleControlData(pModuleType, 1).moduleId
+  return { moduleType = pModuleType, moduleId = pModuleId, subscribe = pSubscribe }
+end
+
+--[[ @interiorVDvalidation: validation function for GetInteriorVehicleData request
+--! @parameters:
+--! pOccurences - actual occurrence of GetInteriorVehicleData requests
+--! pExpectedNumber - expected number of GetInteriorVehicleData requests
+--! pActualData - actual data to compare
+--! pExpectedData - expected data to compare
+--! @return: validation result
+--]]
+function m.interiorVDvalidation(pOccurences, pExpectedNumber, pActualData, pExpectedData)
+  if pOccurences == pExpectedNumber then
+    if m.isTableEqual(pExpectedData, pActualData) == false then
+      local errorMessage = "Wrong expected requests are received\n" ..
+      "Actual result:" .. m.tableToString(pActualData) .. "\n" ..
+      "Expected result:" .. m.tableToString(pExpectedData) .."\n"
+    return false, errorMessage
+    end
+  end
+  return true
```

Comment:
@jacobkeeler Yes we are. This function should return `boolean` value. 
The data check should be performed only once when all responses are received, because their order can be various. 
That is why the function returns `true` without any checks for all responses except last one.